### ISSUE TITLE
memory_order_release idl cxx code generator

### ISSUE THREAD

**Issue by nate800** [eclipse-cyclonedds/cyclonedds-cxx#276]
The code generator produces "std::memory_order::memory_order_release", which will only be part of the C++ standard until C++20. 

https://en.cppreference.com/w/cpp/atomic/memory_order

**Comment by reicheratwork** [comment]
> The code generator produces "std::memory_order::memory_order_release", which will only be part of the C++ standard until C++20.
> 
> https://en.cppreference.com/w/cpp/atomic/memory_order

@nate800  you are correct, seems like the memory order is only an enum class since C++20.
But at the same time, there are mappings of the old (non-scoped) enumerator names to the new scoped values, which should make it very easy to have code generated that is independent on whether the compiler follows C++11 or C++20.

I will reference this issue in a PR which will contain a fix.
